compiler_gym/envs/compiler_env.py

```diff
@@ -269,23 +275,14 @@ def __init__(
             for space in self.service.action_spaces
         ]
         self.observation = self._observation_view_type(
-            get_observation=lambda req: self.service(
-                self.service.stub.GetObservation, req
-            ),
+            get_observation=lambda req: self.service(self.service.stub.Step, req),
             spaces=self.service.observation_spaces,
         )
-        self.reward = self._reward_view_type(
-            get_reward=lambda req: self.service(self.service.stub.GetReward, req),
-            spaces=self.service.reward_spaces,
-        )
+        self.reward = self._reward_view_type(rewards, self.observation)
 
         # Lazily evaluated version strings.
         self._versions: Optional[GetVersionReply] = None
 
-        # A compiler service supports multiple simultaneous environments. This
-        # session ID is used to identify this environment.
-        self._session_id: Optional[int] = None
```

Comment:
Unfortunately not. I think sessions will be useful for breaking the "1 service, 1 environment" requirement (which will make `env.fork()` much faster), but currently don't do much. The reason this line is moved in this diff is only to ensure that `self._session_id` is set early during construction in case of error :)